pennylane/tape/tapes/tape.py

```diff
@@ -220,6 +221,9 @@ class QuantumTape(AnnotatedQueue):
     [0.56, 0.543, 0.133]
     """
 
+    _lock = RLock()
+    """threading.RLock: Used to synchronize appending to/popping from global QueueingContext."""
```

Comment:
Nice!

Reply:
Out of curiosity, how come you use `RLock` here instead of `Lock`? Is there a particular implementation detail below that requires the use of `RLock`?

I'm not so familiar with the `threading` module, so asking for my own curiosity :)

Reply:
Thank you! I see some test was deadlocked if `Lock` is used. Therefore I thought `with` block can be nested. If nested, non re-entrant lock deadlock itself because of his own lock.

Reply:
Oh of course! That answers my question